src/tools/compiletest/src/common.rs

```diff
@@ -651,6 +657,17 @@ pub enum Endian {
     Big,
 }
 
+fn builtin_cfg_names(config: &Config) -> HashSet<String> {
+    rustc_output(
+        config,
+        &["--print=check-cfg", "-Zunstable-options", "--check-cfg=cfg()"],
+        Default::default(),
+    )
+    .lines()
+    .map(|l| if let Some((name, _)) = l.split_once('=') { name.to_string() } else { l.to_string() })
```

Comment:
Just a FWIW, this parsing is quite primitive, it doesn't handle the [`--print=check-cfg` specifics](https://doc.rust-lang.org/nightly/unstable-book/compiler-flags/print-check-cfg.html#printcheck-cfg), fortunately this is only collecting names and shouldn't be in any specific case so it should always work.

Reply:
Yeah, this is only querying for the cfg name, otherwise this can't be used